x/datapool/keeper/pool_distribute.go

```diff
@@ -0,0 +1,227 @@
+	if salesHistory == nil {
+		salesHistory = &types.SalesHistory{
+			PoolId:        poolID,
+			Round:         round,
+			SellerAddress: addr.String(),
+			DataHashes:    [][]byte{dataHash},
+			PaidCoin:      &types.ZeroFund,
+		}
+	} else {
+		salesHistory.AddDataHash(dataHash)
+	}
+
+	k.SetSalesHistory(ctx, salesHistory)
+}
+
+// SetSalesHistory stores sales history.
+func (k Keeper) SetSalesHistory(ctx sdk.Context, salesHistory *types.SalesHistory) {
+	key := types.GetKeyPrefixSalesHistory(
+		salesHistory.PoolId,
+		salesHistory.Round,
+		salesHistory.SellerAddress,
+	)
+	store := ctx.KVStore(k.storeKey)
+	store.Set(key, k.cdc.MustMarshalBinaryLengthPrefixed(salesHistory))
+}
+
+// GetSalesHistory returns the sales history. If there is no value, it responds nil.
+func (k Keeper) GetSalesHistory(ctx sdk.Context, poolID, round uint64, sellerAddress string) *types.SalesHistory {
+	key := types.GetKeyPrefixSalesHistory(poolID, round, sellerAddress)
+	store := ctx.KVStore(k.storeKey)
+	if !store.Has(key) {
+		return nil
+	}
+	bz := store.Get(key)
+	var salesHistory types.SalesHistory
+	k.cdc.MustUnmarshalBinaryLengthPrefixed(bz, &salesHistory)
+	return &salesHistory
+}
+
+func (k Keeper) ListSalesHistory(ctx sdk.Context, poolID, round uint64) []*types.SalesHistory {
+	key := types.GetKeyPrefixSalesHistories(poolID, round)
+	store := ctx.KVStore(k.storeKey)
+	iter := sdk.KVStorePrefixIterator(store, key)
+	var histories []*types.SalesHistory
+	for ; iter.Valid(); iter.Next() {
+		history := &types.SalesHistory{}
+		bz := iter.Value()
+		k.cdc.MustUnmarshalBinaryLengthPrefixed(bz, history)
+		histories = append(histories, history)
+	}
+	return histories
+}
+
+// SetInstantRevenueDistribute stores the poolID to which the revenue should be distributed immediately.
+func (k Keeper) SetInstantRevenueDistribute(ctx sdk.Context, instantRevenueDistribute *types.InstantRevenueDistribute) {
+	bz := k.cdc.MustMarshalBinaryLengthPrefixed(instantRevenueDistribute)
+	store := ctx.KVStore(k.storeKey)
+	store.Set(types.KeyPrefixInstantRevenueDistribute, bz)
+}
+
+func (k Keeper) GetInstantRevenueDistribute(ctx sdk.Context) *types.InstantRevenueDistribute {
+	store := ctx.KVStore(k.storeKey)
+	if !store.Has(types.KeyPrefixInstantRevenueDistribute) {
+		return &types.InstantRevenueDistribute{}
+	}
+	bz := store.Get(types.KeyPrefixInstantRevenueDistribute)
+	var delayedRevenueDistribute types.InstantRevenueDistribute
+	k.cdc.MustUnmarshalBinaryLengthPrefixed(bz, &delayedRevenueDistribute)
+	return &delayedRevenueDistribute
+}
+
+// addInstantRevenueDistribute adds the poolID to distribute. If there are duplicate poolIDs, they are not added.
+func (k Keeper) addInstantRevenueDistribute(ctx sdk.Context, poolID uint64) error {
+	instantRevenueDistribute := k.GetInstantRevenueDistribute(ctx)
+	instantRevenueDistribute.AppendPoolID(poolID)
+	k.SetInstantRevenueDistribute(ctx, instantRevenueDistribute)
+	return nil
+}
```

Comment:
Why this method return `error`? I think that this method has no return values.

Reply:
Yes, you're right.
I fixed it. (https://github.com/medibloc/panacea-core/pull/318/commits/dfe9a6ca02ba0b18b81276c02872e62313815e5f)